core/src/database/migrations.cpp

```diff
@@ -1205,5 +1198,29 @@ namespace ledger {
             sql << "DROP INDEX bitcoin_operations_transaction_uid_index ;";
         }
 
+        template <> void migrate<29>(soci::session& sql, api::DatabaseBackendType type) {
+            sql << "ALTER TABLE bech32_parameters ADD p2trversion VARCHAR(255) DEFAULT 0x01;";
+        }
+
+        template <> void rollback<29>(soci::session& sql, api::DatabaseBackendType type) {
+            // SQLite doesn't handle ALTER TABLE DROP
+            if (type != api::DatabaseBackendType::SQLITE3) {
+                sql << "ALTER TABLE bech32_parameters DROP p2trversion";
+            } else {
+                sql << "CREATE TABLE bech32_parameters_swap("
+                       "name VARCHAR(255) PRIMARY KEY NOT NULL REFERENCES bitcoin_currencies(name) ON DELETE CASCADE ON UPDATE CASCADE,"
+                       "hrp VARCHAR(255) NOT NULL,"
+                       "separator VARCHAR(255) NOT NULL,"
+                       "generator VARCHAR(255) NOT NULL,"
+                       "p2wpkh_version VARCHAR(255) NOT NULL,"
+                       "p2wsh_version VARCHAR(255) NOT NULL"
+                       ")";
+                sql << "INSERT INTO bech32_parameters_swap "
+                       "SELECT name, hrp, separator, generator, p2wpkh_version, p2wsh_version "
+                       "FROM bech32_parameters";
+                sql << "DROP TABLE bech32_parameters";
```

Comment:
Based on your feedback and on my reading, I confirm we should discuss this line and the effect of dropping the table and which effect it could have on cascade deletion over `name`.

Reply:
As discussed with @viktorb-ledger, let's keep the implementation as proposed in this PR for the migration rollback step, considering that SQLite is actively used in tests. A [task has been added](https://ledgerhq.atlassian.net/browse/VG-5378) in the backlog to handle the upgrade of SQLite to support the same syntax than postgres and support dropping a single column.